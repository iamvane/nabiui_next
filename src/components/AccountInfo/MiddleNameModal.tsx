import * as React from 'react';

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  TextField
} from '@material-ui/core';

import {
  AccountInfoComponent,
  MiddleNameModalComponent
} from 'components/AccountInfo/constants';

interface Props {
  isFormDialogOpen: boolean;
  closeHandler: () => void;
  handleChange: (event: React.FormEvent<{}>) => void;
  middleName: string;
  confirmNoMiddleName: () => void;
  applyMiddleName: () => void;
  error?: string;
}

const MiddleNameModal: React.StatelessComponent <Props> = props => {
  const {
    FieldNames,
    FieldKey
  } = AccountInfoComponent;
  return (
    <div>
      <Dialog
        open={props.isFormDialogOpen}
        onClose={props.closeHandler}
        aria-labelledby="form-dialog-title"
        scroll="body"
      >
        <DialogTitle id="form-dialog-title nabi-text-uppercase">{MiddleNameModalComponent.title}</DialogTitle>

        <DialogContent>
          <DialogContentText>
            {MiddleNameModalComponent.message}
            <TextField
              fullWidth={true}
              name={FieldNames[FieldKey.MiddleName]}
              onChange={props.handleChange}
              placeholder={AccountInfoComponent.Placeholders.MiddleName}
              required={true}
              value={props.middleName}
              error={!!props.error}
              helperText={props.error}
            />
          </DialogContentText>
        </DialogContent>

        <DialogActions>
          <Button onClick={props.confirmNoMiddleName}>{MiddleNameModalComponent.noMiddleName}</Button>

          <Button variant="contained" onClick={props.applyMiddleName} color="primary">
            {MiddleNameModalComponent.applyButton}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default MiddleNameModal;
