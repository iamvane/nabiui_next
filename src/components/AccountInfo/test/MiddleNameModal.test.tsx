import * as React from 'react';
import { shallow } from 'enzyme';
import MiddleNameModal from 'components/AccountInfo/MiddleNameModal';

describe('MiddleNameModal', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <MiddleNameModal
        isFormDialogOpen={false}
        closeHandler={jest.fn()}
        handleChange={jest.fn()}
        middleName="middleName"
        confirmNoMiddleName={jest.fn()}
        applyMiddleName={jest.fn()}
      />
    );
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
