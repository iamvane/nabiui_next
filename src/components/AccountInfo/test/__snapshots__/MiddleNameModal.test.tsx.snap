// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiddleNameModal Renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MiddleNameModal
    applyMiddleName={[MockFunction]}
    closeHandler={[MockFunction]}
    confirmNoMiddleName={[MockFunction]}
    handleChange={[MockFunction]}
    isFormDialogOpen={false}
    middleName="middleName"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        aria-labelledby="form-dialog-title"
        onClose={[MockFunction]}
        open={false}
        scroll="body"
      >
        <WithStyles(DialogTitle)
          id="form-dialog-title nabi-text-uppercase"
        >
          Missing Middle Name
        </WithStyles(DialogTitle)>
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)>
            Looks like you didn’t provide a middle name. Please enter it below, if applicable.
            <TextField
              error={false}
              fullWidth={true}
              helperText={undefined}
              name="middleName"
              onChange={[MockFunction]}
              placeholder="Middle Name"
              required={true}
              select={false}
              value="middleName"
              variant="standard"
            />
          </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            onClick={[MockFunction]}
          >
            I don't have one
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[MockFunction]}
            variant="contained"
          >
            Apply
          </WithStyles(Button)>
        </WithStyles(DialogActions)>
      </WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "form-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="form-dialog-title nabi-text-uppercase"
          >
            Missing Middle Name
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              Looks like you didn’t provide a middle name. Please enter it below, if applicable.
              <TextField
                error={false}
                fullWidth={true}
                helperText={undefined}
                name="middleName"
                onChange={[MockFunction]}
                placeholder="Middle Name"
                required={true}
                select={false}
                value="middleName"
                variant="standard"
              />
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              onClick={[MockFunction]}
            >
              I don't have one
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
              variant="contained"
            >
              Apply
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [MockFunction],
        "open": false,
        "scroll": "body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Missing Middle Name",
            "id": "form-dialog-title nabi-text-uppercase",
          },
          "ref": null,
          "rendered": "Missing Middle Name",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)>
              Looks like you didn’t provide a middle name. Please enter it below, if applicable.
              <TextField
                error={false}
                fullWidth={true}
                helperText={undefined}
                name="middleName"
                onChange={[MockFunction]}
                placeholder="Middle Name"
                required={true}
                select={false}
                value="middleName"
                variant="standard"
              />
            </WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Looks like you didn’t provide a middle name. Please enter it below, if applicable.",
                <TextField
                  error={false}
                  fullWidth={true}
                  helperText={undefined}
                  name="middleName"
                  onChange={[MockFunction]}
                  placeholder="Middle Name"
                  required={true}
                  select={false}
                  value="middleName"
                  variant="standard"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Looks like you didn’t provide a middle name. Please enter it below, if applicable.",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "error": false,
                  "fullWidth": true,
                  "helperText": undefined,
                  "name": "middleName",
                  "onChange": [MockFunction],
                  "placeholder": "Middle Name",
                  "required": true,
                  "select": false,
                  "value": "middleName",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                onClick={[MockFunction]}
              >
                I don't have one
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[MockFunction]}
                variant="contained"
              >
                Apply
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "I don't have one",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "I don't have one",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Apply",
                "color": "primary",
                "onClick": [MockFunction],
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Apply",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          aria-labelledby="form-dialog-title"
          onClose={[MockFunction]}
          open={false}
          scroll="body"
        >
          <WithStyles(DialogTitle)
            id="form-dialog-title nabi-text-uppercase"
          >
            Missing Middle Name
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              Looks like you didn’t provide a middle name. Please enter it below, if applicable.
              <TextField
                error={false}
                fullWidth={true}
                helperText={undefined}
                name="middleName"
                onChange={[MockFunction]}
                placeholder="Middle Name"
                required={true}
                select={false}
                value="middleName"
                variant="standard"
              />
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              onClick={[MockFunction]}
            >
              I don't have one
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
              variant="contained"
            >
              Apply
            </WithStyles(Button)>
          </WithStyles(DialogActions)>
        </WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-labelledby": "form-dialog-title",
          "children": Array [
            <WithStyles(DialogTitle)
              id="form-dialog-title nabi-text-uppercase"
            >
              Missing Middle Name
            </WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)>
                Looks like you didn’t provide a middle name. Please enter it below, if applicable.
                <TextField
                  error={false}
                  fullWidth={true}
                  helperText={undefined}
                  name="middleName"
                  onChange={[MockFunction]}
                  placeholder="Middle Name"
                  required={true}
                  select={false}
                  value="middleName"
                  variant="standard"
                />
              </WithStyles(DialogContentText)>
            </WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                onClick={[MockFunction]}
              >
                I don't have one
              </WithStyles(Button)>
              <WithStyles(Button)
                color="primary"
                onClick={[MockFunction]}
                variant="contained"
              >
                Apply
              </WithStyles(Button)>
            </WithStyles(DialogActions)>,
          ],
          "onClose": [MockFunction],
          "open": false,
          "scroll": "body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Missing Middle Name",
              "id": "form-dialog-title nabi-text-uppercase",
            },
            "ref": null,
            "rendered": "Missing Middle Name",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(DialogContentText)>
                Looks like you didn’t provide a middle name. Please enter it below, if applicable.
                <TextField
                  error={false}
                  fullWidth={true}
                  helperText={undefined}
                  name="middleName"
                  onChange={[MockFunction]}
                  placeholder="Middle Name"
                  required={true}
                  select={false}
                  value="middleName"
                  variant="standard"
                />
              </WithStyles(DialogContentText)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Looks like you didn’t provide a middle name. Please enter it below, if applicable.",
                  <TextField
                    error={false}
                    fullWidth={true}
                    helperText={undefined}
                    name="middleName"
                    onChange={[MockFunction]}
                    placeholder="Middle Name"
                    required={true}
                    select={false}
                    value="middleName"
                    variant="standard"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Looks like you didn’t provide a middle name. Please enter it below, if applicable.",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "error": false,
                    "fullWidth": true,
                    "helperText": undefined,
                    "name": "middleName",
                    "onChange": [MockFunction],
                    "placeholder": "Middle Name",
                    "required": true,
                    "select": false,
                    "value": "middleName",
                    "variant": "standard",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Button)
                  onClick={[MockFunction]}
                >
                  I don't have one
                </WithStyles(Button)>,
                <WithStyles(Button)
                  color="primary"
                  onClick={[MockFunction]}
                  variant="contained"
                >
                  Apply
                </WithStyles(Button)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "I don't have one",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "I don't have one",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Apply",
                  "color": "primary",
                  "onClick": [MockFunction],
                  "variant": "contained",
                },
                "ref": null,
                "rendered": "Apply",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
